### ISSUE TITLE
DmassQ input results in the wrong state set

### ISSUE THREAD

**Issue by bryanwweber** [CoolProp/CoolProp#1556]
Hello,

I am trying to fix the state of a substance using the density and quality as inputs. This consistently results in the quality being set equal to 0, regardless of the fluid, density, or quality given as inputs. This occurs for the AbstractState or the PropsSI function. I'm using Python 3.6 and CoolProp 6.1.0.

```
import CoolProp
abst = CoolProp.AbstractState('HEOS', 'water')
abst.update(CoolProp.DmassQ_INPUTS, 1.0, 1.0)
print(abst.keyed_output(CoolProp.iQ))
```
will print `0.0` for any value of `D` that I have tried, provided that `Q` is within some range depending on the density (improper values of Q result in a `ValueError` from the solver, which makes sense). Further, if I use the PropsSI function, it returns the quality as `1.0`, but clearly calculates the value of the energy as the saturated liquid:

```
from CoolProp.CoolProp import PropsSI
print(PropsSI('Q', 'Q', 1.0, 'D', 1.0, 'water'))
print(PropsSI('U', 'Q', 1.0, 'D', 1.0, 'water'))
print(PropsSI('H', 'Q', 1.0, 'D', 1.0, 'water'))
```
prints

```
1.0
487317.6920189324
487503.35652941227
```

However, if I find a separate property (e.g., the pressure) and use that to set the state, it returns the same density as I started with and has the proper internal energy:

```
import CoolProp
abst = CoolProp.AbstractState('HEOS', 'water')
abst.update(CoolProp.DmassQ_INPUTS, 1.0, 1.0)
print(abst.keyed_output(CoolProp.iP))
print(abst.keyed_output(CoolProp.iDmass))
print(abst.keyed_output(CoolProp.iQ))
abst.update(CoolProp.PQ_INPUTS, 175670.3307708434, 1.0)
print(abst.keyed_output(CoolProp.iDmass))
print(abst.keyed_output(CoolProp.iQ))
print(abst.keyed_output(CoolProp.iUmass))
```

prints

```
175670.3307708434
1.0000000000000182
0.0
1.000000000000017
1.0
2524623.713130115
```

Thank you!

**Comment by ibell** [comment]
The problem is that density is multi-valued for the saturated vapor.  I
don't recall how that was handled in CoolProp, but I can look into it.  In
principle, there are two solutions.  Plotting the saturated vapor density
as a function of T will clearly show this "feature"

**Comment by bryanwweber** [comment]
Thank you! Glad to know I'm not losing my mind 😄 I'm curious how the density can be multi-valued for the saturated vapor, except as a numerical oddity, since I would think for a pure substance that the two independent properties completely fix the intensive state. Is this a numerical oddity or am I missing something? If the latter, how does using the pressure and quality select the "correct" density?

**Comment by ibell** [comment]
Hmm, looks like the saturated vapor density for water is not multi-valued.
This seems like a bug in CoolProp.  In general, PQ will be faster and more
reliable for pure fluids.  You should use PQ instead of DQ if you can.

**Comment by bryanwweber** [comment]
This has been fixed in 6.2.1. I'm not sure what commit fixed it, but I'm going to close the issue. Thank you!
